src/ctelnet.cpp

```diff
@@ -1413,6 +1467,72 @@ void cTelnet::processTelnetCommand(const std::string& command)
     case TN_SB: {
         option = command[2];
 
+        // CHARSET
+        if (option == OPT_CHARSET && enableCHARSET) {
+            QByteArray payload = command.c_str();
+            if (payload.size() < 6) {
+                return;
+            }
+
+            // Trim off the Telnet suboption bytes from beginning (3) and end (2)
+            payload = payload.mid(3, static_cast<int>(payload.size()) - 5);
+
+            // CHARSET support per https://tools.ietf.org/html/rfc2066
+            if (command[3] == CHARSET_REQUEST) {
+                if (payload.startsWith("[TTABLE]1")) { // No translate table support.  Discard.
+                    payload.remove(0, 9);
+                }
+
+                QList<QByteArray> characterSetList = payload.split(payload[1]); // Second character is the separator.
+                QByteArray acceptedCharacterSet;
+
+                if (characterSetList.size() > 0) {
+                    for (int i = 1; i < characterSetList.size(); ++i) {
+                        if (mAcceptableEncodings.contains(characterSetList[i]) || mAcceptableEncodings.contains(("M_" + characterSetList[i]))) {
```

Comment:
:information_source: Don't use `operator[`n`]` on `QList<T>` containers unless you **need** to modify the contents - the `const QList<T>::at(`n`)` is faster and recommend in the read-only access cases. Also all our encoding names ARE UPPERCASE - so it might be sensible to use `characterSetList.at(i).toUpper()` in the two places here...